front50-core/src/main/groovy/com/netflix/spinnaker/front50/model/plugins/PluginInfoService.java

```diff
@@ -103,6 +104,28 @@ public PluginInfo deleteRelease(@Nonnull String id, @Nonnull String releaseVersi
     return pluginInfo;
   }
 
+  /** Set the preferred release. If preferred is true, sets previous preferred release to false. */
+  public PluginInfo.Release preferReleaseVersion(
+      @Nonnull String id, @Nonnull String releaseVersion, boolean preferred) {
+    PluginInfo pluginInfo = repository.findById(id);
+    Optional<PluginInfo.Release> release = pluginInfo.getReleaseByVersion(releaseVersion);
+
+    if (release.isPresent()) {
+      release.get().setPreferred(preferred);
+      pluginInfo.setReleaseByVersion(releaseVersion, release.get());
+
+      if (preferred) {
+        pluginInfo.getReleases().stream()
+            .filter(it -> !it.getVersion().equals(release.get().getVersion()))
+            .forEach(it -> it.setPreferred(false));
+      }
+
+      repository.update(pluginInfo.getId(), pluginInfo);
+    }
+
+    return release.orElse(null);
```

Comment:
nit: `release.map(r -> { /* do your stuff */).orElse(null)` -- then you can avoid the `isPresent()` and explicit `get()` call.